src/main/scala/net/cakesolutions/CakeBuildPlugin.scala

```diff
@@ -32,23 +42,33 @@ object CakeBuildPlugin extends AutoPlugin {
       case _             => List("-XX:MaxMetaspaceSize=256m")
     }
 
-  override val buildSettings = Seq(
+  /** @inheritdoc */
+  override val buildSettings: Seq[Setting[_]] = Seq(
     organization := "net.cakesolutions",
     scalaVersion := "2.11.11",
     maxErrors := 1,
     fork := true,
     cancelable := true,
     sourcesInBase := false,
-    javaOptions += s"-Dcake.sbt.root=${(baseDirectory in ThisBuild).value.getCanonicalFile}",
+    javaOptions +=
+      s"-Dcake.sbt.root=${(baseDirectory in ThisBuild).value.getCanonicalFile}",
     // WORKAROUND DockerPlugin doesn't like '+'
     version := version.value.replace('+', '-'),
     concurrentRestrictions := {
-      val limited = Properties.envOrElse("SBT_TASK_LIMIT", "4").toInt
+      val limited = sys.env
+        .get("SBT_TASK_LIMIT")
+        .flatMap(n => Try(n.toInt).toOption)
```

Comment:
silent failure is not good here, we want an exception if the user provides a non-integer value